src/Normalizer/UnixMillisecondNormalizer.php

```diff
@@ -5,24 +5,69 @@
 use Lullabot\Mpx\DataService\DateTime\ConcreteDateTime;
 use Lullabot\Mpx\DataService\DateTime\ConcreteDateTimeInterface;
 use Lullabot\Mpx\DataService\DateTime\DateTimeFormatInterface;
+use Symfony\Component\PropertyInfo\Type;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
 use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
-use Symfony\Component\Serializer\Normalizer\DateTimeNormalizer;
+use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
 
 /**
  * Denormalize a millisecond timestamp into a date object.
+ *
+ * Note this doesn't extend the
+ * \Symfony\Component\Serializer\Normalizer\DateTimeNormalizer normalizer
+ * directly because the ::denormalize() method in that class is typehinted to
+ * \DateTimeInterface and we want to return a ConcreteDateTime object instead.
+ * Unfortunately \DateTimeInterface can't be extended by user classes, so the
+ * option we have is to unfortunately copy here most methods from
+ * DateTimeNormalizer, and modify the ones that are particular to us here.
```

Comment:
This is unfortunately the only way I could find to workaround the fact that we can't extend `\DateTimeInterface` but I would be happy to go into another direction if others can think of alternative approaches.

Reply:
Would it be possible to use a [decorator pattern](https://en.wikipedia.org/wiki/Decorator_pattern) here? I think this is a perfect use case for it.

Reply:
Good call! I'll give it a try. Thanks!